src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+     * @private @const {!Object<string,boolean>}
+     */
+    this.topLevelParams_ = {
+      'client_id': true,
+      'user_id': true,
+      'timestamp_micros': true,
+      'user_properties': true,
+      'non_personalized_ads': true,
+    };
+
+    /**
+     * An API secret that can be generated via Google Analytics UI
+     * NOTE: This processor makes public network requests using this
+     * API secret. To avoid spam, regenerate API secrets regularly.
+     * @private @const {string|undefined}
+     */
+    this.apiSecret_ = apiSecret;
+
+    /**
+     * The identifier for property being measured. Can be found using
+     * the Google Analytics UI.
+     * @private @const {string|undefined}
+     */
+    this.measurementId_ = measurementId;
+
+    /**
+     * URL to send network requests to. Defaults to the Google Analytics
+     * collection endpoint. Data must be sent via HTTPS.
+     * @private @const {string}
+     */
+    this.measurementUrl_ = measurementUrl;
+
+    /**
+     * How long the client ID key should be stored in long term storage
+     * measured in seconds. Defaults to positive infinity.
+     * @private @const {number}
+     */
+    this.clientIdExpires_ = clientIdExpires;
+  }
+
+  /**
+   * Names the event processor so multiple processors
+   * can share previous configuration.
+   * @return {string}
+   * @export
+   */
+  getName() {
+    return 'googleAnalytics';
+  }
+
+  /**
+   * Processes events pushed to the data layer by constructing and sending JSON
+   * POST requests to Google Analytics. Follows Measurement Protocol.
+   * @param {!StorageInterface} storageInterface An interface to an object to
+   *    load or save persistent data with.
+   * @param {{get:function(string):*, set:function(string, *)}} modelInterface
+   *    An interface to load or save short term page data from the data layer.
+   * @param {string} eventName The name of the event passed to the data layer.
+   * @param {!Object<string, *>=} eventArgs The events passed to the data layer.
+   * @export
+   */
+  processEvent(storageInterface, modelInterface, eventName, eventArgs) {
+    // constructs and sends JSON POST requests to GA
+  }
+
+  /**
+   * Checks if the key being set by the user is client ID and if so returns
+   * the client ID expiry.
+   * @param {string} key The location at which to store data in the model.
+   *    Dot notation is used to access a nested value (i.e. 'employees.jim'
+   *    is the key 'jim' in the nested 'employees' object).
+   * @param {*} value The data to store.
+   * @return {number} How long the key should be stored in seconds.
+   *     If -1, then the default value saved in storage will be used.
+   *     If 0, the data is not saved to long term storage at all.
+   *     If Number.POSITIVE_INFINITY, data will be stored forever.
+   * @export
+   */
+  persistTime(key, value) {
+    // checks to see if key is client ID
+    // Returns client ID expires if true
+    return -1;
```

Comment:
This comment is inaccurate.

Reply:
+1.  If it's a TODO comment, add TODO(kjgalvan): at the start.

Reply:
Made this explicitly a TODO